### PR TITLE
Migrate to Scala 3 syntax

### PR DESCRIPTION
**Pull request by armanbilge**

### CODE REVIEW

File: core/src/test/scala/org/typelevel/discipline/laws.scala [hunk 1]
```diff
@@ -23,13 +23,14 @@ package org.typelevel.discipline
 
 import org.scalacheck.Prop
 
-object Dummy {
+import scala.language.adhocExtensions
```
Comment (vasilmkd) [inline core/src/test/scala/org/typelevel/discipline/laws.scala:26]: Why exactly is this import necessary?

Comment (armanbilge) [reply core/src/test/scala/org/typelevel/discipline/laws.scala:26]: Good point, actually I don't think it is. Just have to mark the classes `open`.

Comment (armanbilge) [reply core/src/test/scala/org/typelevel/discipline/laws.scala:26]:
```scala
[warn] -- Feature Warning: /workspace/discipline-specs2/core/src/test/scala/org/typelevel/discipline/laws.scala:70:12 
[warn] 70 |  ) extends DefaultRuleSet(name, parent, props*)
[warn]    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[warn]    |Unless class DefaultRuleSet is declared 'open', its extension in a separate file should be enabled
```

Comment (armanbilge) [reply core/src/test/scala/org/typelevel/discipline/laws.scala:26]: Actually we do need it then I think.

Comment (vasilmkd) [reply core/src/test/scala/org/typelevel/discipline/laws.scala:26]: `open` doesn't work?

Comment (armanbilge) [reply core/src/test/scala/org/typelevel/discipline/laws.scala:26]: `DefaultRuleSet` is not defined here so we can't change it.

Comment (vasilmkd) [reply core/src/test/scala/org/typelevel/discipline/laws.scala:26]: That comes from `specs2`?

Comment (armanbilge) [reply core/src/test/scala/org/typelevel/discipline/laws.scala:26]: No, it comes from discipline.
https://github.com/typelevel/discipline/blob/79a121e3e0c605bcd30f73fa3f88830b521e35c4/core/src/main/scala/org/typelevel/discipline/Laws.scala#L107

Comment (vasilmkd) [reply core/src/test/scala/org/typelevel/discipline/laws.scala:26]: Cool. Thanks.

**Review by vasilmkd — APPROVED** [review]

**Pull request merged**